### PR TITLE
[prompty] prompty support sample file as input

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
- json file as sample input
```
---
name: Basic Prompt
description: A basic prompt that uses the GPT-3 chat API to answer questions
model:
    api: chat
    configuration:
      type: azure_openai
      azure_deployment: gpt-35-turbo
      connection: azure_open_ai_connection
    parameters:
      max_tokens: 128
      temperature: 0.2
inputs:
  firstName:
    type: string
    default: John
  lastName:
    type: string
    default: Doh
  question:
    type: string
sample: path/to/inputs.json
---
system:
You are an AI assistant who helps people find information.
As the assistant, you answer questions briefly, succinctly,
and in a personable manner using markdown and even add some personal flair with appropriate emojis.

# Safety
- You **should always** reference factual statements to search results based on [relevant documents]
- Search results based on [relevant documents] may be incomplete or irrelevant. You do not make assumptions
# Customer
You are helping {{firstName}} {{lastName}} to find answers to their questions.
Use their name to address them in your responses.

user:
{{question}}

```

- dict as sample input
```
---
name: Basic Prompt
description: A basic prompt that uses the GPT-3 chat API to answer questions
model:
    api: chat
    configuration:
      type: azure_openai
      azure_deployment: gpt-35-turbo
      connection: azure_open_ai_connection
    parameters:
      max_tokens: 128
      temperature: 0.2
inputs:
  firstName:
    type: string
    default: John
  lastName:
    type: string
    default: Doh
  question:
    type: string
sample:
  question: what is the result of 1+1?
---
system:
You are an AI assistant who helps people find information.
As the assistant, you answer questions briefly, succinctly,
and in a personable manner using markdown and even add some personal flair with appropriate emojis.

# Safety
- You **should always** reference factual statements to search results based on [relevant documents]
- Search results based on [relevant documents] may be incomplete or irrelevant. You do not make assumptions
# Customer
You are helping {{firstName}} {{lastName}} to find answers to their questions.
Use their name to address them in your responses.

user:
{{question}}

```

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: docs/reference/pf-command-reference.md [hunk 1]
```diff
@@ -112,6 +112,18 @@ pf flow test --flow <path-to-flow-directory>
 
 Test the flow with single line from input file.
 
+```bash
+pf flow test --flow <path-to-flow-directory> --inputs inputs.jsonl
+```
+
+Test the flow from json file.
```
Comment (wangchao1230) [inline docs/reference/pf-command-reference.md:119]: put this the first ahead of the jsonl one

File: docs/reference/pf-command-reference.md [hunk 2]
```diff
@@ -112,6 +112,18 @@ pf flow test --flow <path-to-flow-directory>
 
 Test the flow with single line from input file.
 
+```bash
```
Comment (wangchao1230) [inline docs/reference/pf-command-reference.md:115]: single -> first line?

File: src/promptflow/tests/test_configs/prompty/prompty_example_with_sample.prompty [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+inputs:
+  firstName:
+    type: string
+    default: John
+  lastName:
+    type: string
+    default: Doh
+  question:
+    type: string
+sample:
```
[the remark this answers is not in the document]
Comment (lalala123123) [reply src/promptflow/tests/test_configs/prompty/prompty_example_with_sample.prompty:22]: There is a test case will overwrite value of sample into a file.

File: docs/reference/pf-command-reference.md (hunk 1 above)
Comment (lalala123123) [reply docs/reference/pf-command-reference.md]: fix it

File: docs/reference/pf-command-reference.md [hunk 4]
```diff
@@ -112,6 +112,18 @@ pf flow test --flow <path-to-flow-directory>
...
+pf flow test --flow <path-to-flow-directory> --inputs inputs.jsonl
+```
+
+Test the flow from json file.
+
+```bash
+pf flow test --flow <path-to-flow-directory> --inputs inputs.json
```
[the remark this answers is not in the document]
Comment (lalala123123) [reply docs/reference/pf-command-reference.md]: do you mean add sample in flex.yaml
```
entry: module:my_func
sample: path/to/sample.json
```

File: src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_prompty.py [hunk 5]
```diff
@@ -251,3 +256,28 @@ def test_prompty_trace(self, pf: PFClient):
         events = [pf.traces.get_event(item["attributes"]["event.id"]) for item in prompty_span.events]
         assert any(["prompt.template" in event["attributes"]["payload"] for event in events])
         assert any(["prompt.variables" in event["attributes"]["payload"] for event in events])
+
+    def test_prompty_with_sample(self, pf: PFClient):
+        prompty = Prompty.load(source=f"{PROMPTY_DIR}/prompty_example_with_sample.prompty")
```
Comment (crazygao) [inline src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_prompty.py:261]: do we support load_flow?
Is it suggested way to import Prompty using
from promptflow.core._flow import AsyncPrompty, Prompty ?

Comment (crazygao) [reply src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_prompty.py:261]: also add some simple cli test.

File: src/promptflow-core/promptflow/core/_prompty_utils.py [hunk 6]
```diff
@@ -43,6 +46,20 @@ def parse_environment_variable(value):
...
+        return {}
+    elif isinstance(sample, dict):
+        return sample
+    elif isinstance(sample, (str, Path)) and Path(sample).suffix.lower() == ".json":
+        if not Path(sample).exists():
+            raise InvalidSampleError(f"Cannot find sample file {sample}.")
+        with open(sample, "r") as f:
+            return json.load(f)
+    else:
+        raise InvalidSampleError("Only dict and json file are supported as sample in prompty.")
```
Comment (crazygao) [inline src/promptflow-core/promptflow/core/_prompty_utils.py:60]: Test this error

Comment (lalala123123) [reply src/promptflow-core/promptflow/core/_prompty_utils.py:60]: There is a test case for this negative scenario.

File: src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_prompty.py (hunk 5 above)
Comment (lalala123123) [reply src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_prompty.py]: The current recommended usage is, Flow.load(source="path/to/prompty.prompty")

Comment (lalala123123) [reply src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_prompty.py]: Add cli test cases.

**Review by wangchao1230 — DISMISSED** [review]

File: docs/reference/pf-command-reference.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (wangchao1230) [reply docs/reference/pf-command-reference.md]: yes.

File: src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_prompty.py (hunk 5 above)
Comment (lalala123123) [reply src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_prompty.py]: we will support load_flow in future

File: docs/reference/pf-command-reference.md [hunk 7]
```diff
@@ -110,7 +110,19 @@ Test the flow.
...
 ```
 
-Test the flow with single line from input file.
+Test the flow from json file.
+
+```bash
+pf flow test --flow <path-to-flow-directory> --inputs inputs.json
+```
+
+Test the flow with first line from input file.
```
[the remark this answers is not in the document]
Comment (lalala123123) [reply docs/reference/pf-command-reference.md]: fix it

File: docs/reference/pf-command-reference.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (lalala123123) [reply docs/reference/pf-command-reference.md]: will support flex flow in next PR

**Review by wangchao1230 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request merged**